server/api/race.js

```diff
@@ -0,0 +1,89 @@
+const router = require('express').Router()
+const {Race, Candidate} = require('../db/models')
+module.exports = router
+
+// GET all races
+router.get('/', async (req, res, next) => {
+  try {
+    const allRaces = await Race.findAll()
+    res.status(200).json(allRaces)
+  } catch (error) {
+    next(error)
+  }
+})
+
+// GET by raceId
+router.get('/:raceId', async (req, res, next) => {
+  try {
+    const particularRace = await Race.findById(+req.params.raceId, {
+      include: [{model: Candidate}]
+    })
+    if (!particularRace) {
+      res.sendStatus(404)
+    } else {
+      res.status(200).json(particularRace)
+    }
+  } catch (error) {
+    next(error)
+  }
+})
+
+// POST a race
+router.post('/', async (req, res, next) => {
+  try {
+    const newRace = await Race.create(req.body)
+    res.status(201).json(newRace)
+  } catch (error) {
+    next(error)
+  }
+})
+
+// PUT by raceId
+router.put('/:raceId', async (req, res, next) => {
+  try {
+    const particularRace = await Race.findById(+req.params.raceId, {
+      include: [{model: Candidate}]
+    })
+    if (!particularRace) {
+      res.sendStatus(404)
+    } else {
+      const updatedRace = particularRace.update(req.body)
+      res.status(201).json(updatedRace)
+    }
+  } catch (error) {
+    next(error)
+  }
+})
+
+// POST candidate by raceId
```

Comment:
this post route needs to be in agreement with route below; if we update at PUT /:raceId/candidates/:candidateId we need to create at POST /:raceId/candidates